### DISCUSSION TITLE
How to access Context values in the application that were set in the agent via a custom TextMapPropagator?

### DISCUSSION THREAD

**Discussion by brenlong83** [open-telemetry/community#1847] [Community Discussions]
We have implemented a custom `TextMapPropagator` as a Java agent extension to propagate an access token across several micro services.  That all works as expected.

Now the interesting part, in some of the micro services we need to introspect the token.

The problem we are facing is the `ContextKey` doesn't match between the app definition and the agent definition as `ContextKey` is compared for equality by reference.  Since the class is effectively duplicated in two class loaders (app/agent) the references don't match so from the application perspective the key is never set.  It's frustrating because if we log the `Context` object we see the token right there, only it's not accessible.

What would be the recommended method to get app access to something that was originally set in the `Context` from the agent?

**Comment by trask** [comment]
hi @brenlong83! I'd recommend asking this question in https://github.com/open-telemetry/opentelemetry-java-instrumentation
